targets/esp8266/user/jerry_run.c

```diff
@@ -27,25 +28,13 @@ jerry_value_t parsed_res;
 
 /*---------------------------------------------------------------------------*/
 
-int js_entry (const char *source_p, const size_t source_size)
+void js_entry ()
 {
-  const jerry_char_t *jerry_src = (const jerry_char_t *) source_p;
-  int ret_code = 0; /* JERRY_COMPLETION_CODE_OK */
+  srand ((unsigned) jerry_port_get_current_time());
   jerry_init_flag_t flags = JERRY_INIT_EMPTY;
 
   jerry_init (flags);
-
   js_register_functions ();
-
-  parsed_res = jerry_parse ((jerry_char_t *) jerry_src, source_size, false);
```

Comment:
Thanks for fixing up `js_eval`. It would be great if `js_loop` could be touched a bit, too. A lot of variable declarations could be (and should be, IMO) moved closer to their first use (and potentially merged with their initialization). (Especially `val_args` could be initialised in a simpler way: `jerry_value_t val_args[] = { jerry_create_number (ticknow) };`, and instead of the current `val_argv` we could have `sizeof (val_args) / sizeof (val_args[0])` or `sizeof (val_args) / sizeof (jerry_value_t)`, which would be more maintainable.)